File: apps/party-time-backend/src/main/java/com/partytime/jpa/mapper/EventParticipantMapper.java
package com.partytime.jpa.mapper;

import com.partytime.api.dto.event.ParticipantDTO;
import com.partytime.jpa.entity.EventParticipant;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class EventParticipantMapper {

    public static ParticipantDTO map(EventParticipant event) {
        return ParticipantDTO.builder()
            .account(AccountMapper.map(event.getAccount()))
            .status(event.getStatus())
            .build();
    }

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/jpa/mapper/EventParticipantMapper.kt
package com.partytime.jpa.mapper

import com.partytime.api.dto.event.ParticipantDTO
import com.partytime.api.dto.event.ParticipatingEventDTO
import com.partytime.jpa.entity.EventParticipant

/**
 * Transforms an [EventParticipant] entity into a (serializable) [ParticipantDTO]
 *
 * @receiver transformation subject
 * @return transformation result, ready for serialization and transmission to the client
 */
fun EventParticipant.toParticipantDTO(): ParticipantDTO = ParticipantDTO(
    account.toAccountDTO(),
    status
)

/**
 * Transforms an [EventParticipant] entity into a (serializable) [ParticipatingEventDTO]
 *
 * @receiver transformation subject
 * @return transformation result, ready for serialization and transmission to the client
 */
fun EventParticipant.toParticipatingEventDTO(): ParticipatingEventDTO = ParticipatingEventDTO(
    event.id!!,
    event.name,
    event.organizer.toAccountDTO(),
    event.dateTime,
    event.address.toAddressDTO(),
    emptyList(),
    status
)
